packages/vats/src/types.js

```diff
@@ -43,3 +43,44 @@
  * @property {NameHub} nameHub
  * @property {NameAdmin} nameAdmin
  */
+
+/**
+ * @typedef {Object} FeeCollector
+ *
+ * @property {() => ERef<Payment>} collectFees
+ */
+
+/**
+ * @typedef {Object} Bank
+ *
+ * @property {(accounts: string[], payments: Payment[]) => void} depositMultiple
+ * @property {() => Notifier<string[]>} getAccountsNotifier
+ */
+
+/**
+ * @typedef {Object} DistributorParams
+ *
+ * @property {number} depositsPerUpdate - (number) how many payments should be
+ *  sent to the Bank interface per updateInterval
+ * @property {bigint} updateInterval - (bigint) parameter to the timer
+ *  controlling the interval at which deposits are sent to the Bank API
+ * @property {bigint} epochInterval - parameter to the epochTimer controlling
+ *  the interval at which rewards should be sent to the bank.
+ * @property {Issuer} runIssuer
+ * @property {Brand} runBrand
+ */
+
+/**
+ * @callback BuildFeeDistributor
+ *
+ * @param {FeeCollector} treasuryCollector - an object with a collectFees()
+ *  method, which will return a payment. can be populated with
+ *  makeTreasuryFeeCollector(zoe, treasuryCreatorFacet)
+ * @param {Bank} bank - object with getAccountsNotifier() and depositMultiple()
+ * @param {TimerService} epochTimer - timer that notifies at the end of each
+ *  Epoch. The epochInterval parameter controls the interval.
+ * @param {TimerService} timer - timer controlling frequency at which batches
```

Comment:
Using `ERef<T>` will help ensure your distributor uses `E(T)[method)(...args)` instead of a synchronous call:

```suggestion
 * @param {ERef<BankDepositFacet>} bank - object with getAccountsNotifier() and depositMultiple()
 * @param {ERef<TimerService>} epochTimer - timer that notifies at the end of each
 *  Epoch. The epochInterval parameter controls the interval.
 * @param {ERef<TimerService>} timer - timer controlling frequency at which batches
```

Reply:
done